src/clr-addons/slider/slider.ts

```diff
@@ -0,0 +1,169 @@
+  @ViewChild('rangeSliderComponent') private _rangeSliderComponent: ClrRange;
+
+  // ====== Events ======
+  valueChanged(value): void {
+    this.onValueChanged.emit(value);
+    switch (value.valueType) {
+      case SliderValueType.value:
+        this._value = value.value;
+        break;
+      case SliderValueType.lowvalue:
+        this._lowValue = +value.value.split(',')[0];
+        break;
+      case SliderValueType.highvalue:
+        this._highValue = +value.value.split(',')[1];
+        break;
+      default:
+        break;
+    }
+  }
+
+  onMinValueChange(newMinValue: number): void {
+    this._minValue = newMinValue;
+  }
+
+  onMaxValueChange(newMaxValue: number): void {
+    this._maxValue = newMaxValue;
+  }
+
+  // == Private Fields ==
+  private _highValue = 50;
+  private _minValue = 0;
+  private _lowValue = 0;
+  private _maxValue = 50;
+  private _step = 1;
+  private _value = 0;
+  private _multiValue = false;
+  private _showsLabels = true;
+  private _enableValueFields = true;
+  private _size = SliderSize.medium;
+
+  // ====== Getter ======
+  public get value(): number {
+    return this._value;
+  }
+
+  public get minValue(): number {
+    return this._minValue;
+  }
+
+  public get maxValue(): number {
+    return this._maxValue;
+  }
+
+  public get step(): number {
+    return this._step;
+  }
+
+  public get showsLabels(): boolean {
+    return this._showsLabels;
+  }
+
+  public get sliderSize(): SliderSize {
+    return this._size;
+  }
+
+  public get valueFieldsAreEnabled(): boolean {
+    return this._enableValueFields;
+  }
+
+  public get multiValue(): boolean {
+    return this._multiValue;
+  }
+
+  public get rangeSliderComponent(): ClrRange {
+    return this._rangeSliderComponent;
+  }
+
+  // ====== Setter ======
+  @Input('clrValue')
+  public set value(value: number) {
+    const newvalue = Math.min(Math.max(value, this._minValue), this._maxValue);
+
+    if (newvalue === this._value) {
+      return;
+    }
+
+    this._value = newvalue;
+    this.onValueChanged.emit({ valueType: SliderValueType.value, value: value });
+  }
+
+  @Input('clrMinValue')
+  public set minValue(min: number) {
+    if (!this._enableValueFields) {
+      return;
+    }
+
+    if (min > this._maxValue) {
+      this._minValue = this._maxValue;
+      this._maxValue = min;
+    } else {
+      this._minValue = min;
+    }
+
+    this.updateValue();
+  }
+
+  @Input('clrMaxValue')
+  public set maxValue(max: number) {
+    if (!this._enableValueFields) {
+      return;
+    }
+
+    if (max < this._minValue) {
+      this._maxValue = this._minValue;
+      this._minValue = max;
+    } else {
+      this._maxValue = max;
+    }
+
+    this.updateValue();
+  }
+
+  @Input('clrStep')
+  public set step(step: number) {
+    this._step = step > 0 ? step : 1;
+  }
+
+  @Input('clrShowsLabels')
+  public set showLabels(showsLabels: boolean) {
```

Comment:
if you dont have additional logic in your setter, give the input to the variable directly.

also remove all of the unused getters and bind to the variables directly, methods like this.updateValue() should not be needed, bindings of angular must take care of this.

Reply:
I think I need to update my values manually as some properties only configure the browser-internal validator.